docs/developer/contributing.md

```diff
@@ -0,0 +1,83 @@
+### Contributing to Pumpkin
+We appreciate your interest in contributing to Pumpkin! This document outlines the guidelines for submitting bug reports, feature suggestions, and code changes.
+
+### Getting Started
+The easiest way to get started is by asking for help in our [discord](https://discord.gg/wT8XjrjKkf).
+
+### How to Contribute
+There are several ways you can contribute to Pumpkin:
+
+#### Reporting Bugs
+  If you encounter a bug, please search for existing issues on the issue tracker first.
+
+  If you can't find a duplicate issue, open a new one.
+
+  Follow the template and provide a clear description of the bug, including steps to reproduce it if possible.
+  Screenshots, logs, or code snippets can also be helpful.
+
+#### Suggesting Features
+  Do you have an idea on how Pumpkin can be improved? Share your thoughts by opening an issue on the issue tracker.
+
+  Describe the proposed feature in detail, including its benefits and potential implementation considerations.
+
+#### Contributing Code
+  To get started with contributing code to Pumpkin, fork the repository on github
+
+- First create a github account if you don't already have one
+ 
+- Go to Pumpkin's Official [Github Repository](https://github.com/Snowiiii/Pumpkin) and press fork
+
+> Creating a fork means you now have your own copy of the Pumpkin source code, (this does not mean you own the copyright)
+
+  Now that you have a copy that you can edit, you will need a few tools
+
+- Install [Github-Desktop](https://desktop.github.com/download/) or [git](https://git-scm.com/downloads) for your operating system
+
+> Github Desktop is a much easier tool to use. If you are unsure of which to use, choose this.
+
+- Install Rust at [rust-lang.org](https://www.rust-lang.org/).
+
+- Finally, if you are wanting to contribute to documentation, install [NodeJS](https://nodejs.org/en)
```

Comment:
I would split this up. If you only want to contribute to docs, you only need node. If you only want to change rust code you only need rust